artemis-console-extension/artemis-extension/src/artemis-extension/artemis/brokers/BrokerTopology.tsx

```diff
@@ -0,0 +1,671 @@
+      return withDndDrop(groupDropTargetSpec)(withDragNode(nodeDragSourceSpec('group'))(withSelection()(DefaultGroup)));
+    default:
+      switch (kind) {
+        case ModelKind.graph:
+          return withDndDrop(graphDropTargetSpec())(withPanZoom()(GraphComponent));
+        case ModelKind.node:
+          switch(type) {
+            case 'broker':
+              return withDndDrop(nodeDropTargetSpec([CONNECTOR_TARGET_DROP]))(
+                withDragNode(nodeDragSourceSpec('node', true, true))(BrokerCustomNode));
+            case 'address':
+              return withDndDrop(nodeDropTargetSpec([CONNECTOR_TARGET_DROP]))(
+                withDragNode(nodeDragSourceSpec('node', true, true))(AddressCustomNode));
+            case 'queue':
+              return withDndDrop(nodeDropTargetSpec([CONNECTOR_TARGET_DROP]))(
+                withDragNode(nodeDragSourceSpec('node', true, true))(QueueCustomNode));
+            case 'internalAddress':
+              return withDndDrop(nodeDropTargetSpec([CONNECTOR_TARGET_DROP]))(
+                withDragNode(nodeDragSourceSpec('node', true, true))(InternalAddressCustomNode));
+            case 'internalQueue':
+              return withDndDrop(nodeDropTargetSpec([CONNECTOR_TARGET_DROP]))(
+                withDragNode(nodeDragSourceSpec('node', true, true))(InternalQueueCustomNode));
+            default:
+              return withDndDrop(nodeDropTargetSpec([CONNECTOR_TARGET_DROP]))(
+                withDragNode(nodeDragSourceSpec('node', true, true))(CustomNode));
+          }
+        case ModelKind.edge:
+          return withTargetDrag<
+            DragObjectWithType,
+            Node,
+            { dragging?: boolean },
+            {
+              element: Edge;
+            }
+          >({
+            item: { type: CONNECTOR_TARGET_DROP },
+            begin: (monitor, props) => {
+              props.element.raise();
+              return props.element;
+            },
+            drag: (event, monitor, props) => {
+              props.element.setEndPoint(event.x, event.y);
+            },
+            end: (dropResult, monitor, props) => {
+              if (monitor.didDrop() && dropResult && props) {
+                props.element.setTarget(dropResult);
+              }
+              props.element.setEndPoint();
+            },
+            collect: monitor => ({
+              dragging: monitor.isDragging()
+            })
+            // @ts-ignore
+          })(DefaultEdge);
+        default:
+          return undefined;
+      }
+  }
+};
+
+const BROKER_NODE_DIAMETER = 75;
+const ADDRESS_NODE_DIAMETER = 50;
+const QUEUE_NODE_DIAMETER = 50;
+
+const NODES: NodeModel[] = [];
+const EDGES: EdgeModel[] = [];
+
+interface ViewOptions {
+  showLabels: boolean;
+  showAddresses: boolean;
+  showQueues: boolean;
+  showInternalAddresses: boolean;
+  showInternalQueues: boolean;
+  showConnectors: boolean;
+}
+
+export const DefaultViewOptions: ViewOptions = {
+  showLabels: false,
```

Comment:
I found it a bit off that the labels are hidden by default. I didnt really know what anything was at all as a result, not until I enabled the labels from the Node Options, since there is no hover-over detail or legend either.  Just some coloured circles. I could guess at the broker one but that was it.

Reply:
I have enabled labels by default.